linodecli/api_request.py

```diff
@@ -65,6 +96,37 @@ def do_request(
     return result
 
 
+@staticmethod
+def _merge_results_data(results: Iterable[dict]):
+    """Merge multiple json response into one"""
+
+    iterator = iter(results)
+    merged_result = next(iterator, None)
+    if not merged_result:
+        return None
+
+    if "pages" in merged_result:
+        merged_result["pages"] = 1
+    if "page" in merged_result:
+        merged_result["page"] = 1
+    if "data" in merged_result:
+        merged_result["data"] += list(
+            itertools.chain.from_iterable(r["data"] for r in iterator)
+        )
+    return merged_result
+
+
+def _get_all_pages_results_generator(
+    ctx,
```

Comment:
Can we add type annotation to `ctx`?

Reply:
Type annotation requires a circular dependency I think. I added comments to clarify though.